src/python/WMComponent/TaskArchiver/TaskArchiverPoller.py

```diff
@@ -690,6 +689,16 @@ def archiveWorkflowSummary(self, spec):
             workflowData['output'][dataset]['events'] = entry['events']
             workflowData['output'][dataset]['tasks']  = outputList.get(dataset, {}).keys()
 
+        # If the workflow was aborted, then don't parse all the jobs, cut at 5k
+        try:
+            reqDetails = self.centralCouchDBWriter.getRequestByNames(workflowName)
+            wfStatus = reqDetails[workflowName]['RequestTransition'][-1]['Status']
+            if wfStatus in ["aborted", "aborted-completed", "aborted-archived"]:
+                logging.info("Workflow %s in status %s with a total of %d jobs, capping at 5000" % (
+                    workflowName, wfStatus, len(failedJobs)))
+                failedJobs = failedJobs[:5000]
```

Comment:
There were 2M jobs in that case, which would take 5 days, at least. I don't remember now how long does it take to process 250 (standard slice size), but I would guess a minute.
I just wanted to be conservative with this 5k jobs, but if you think we can go down, I can set it to 1k jobs only (?) :)